### DISCUSSION TITLE
Combining useSpring with useTransition

### DISCUSSION THREAD

**Discussion by magdalko** [pmndrs/react-spring#1607] [Support]
Hello, 

Does anyone have an idea how to make two different motions on the same component? I have a button which have background color, it changes to a blue color when hovering, but also changes to darker blue when clicking on it. I did both on hover and on click separately in different examples (for hover I used useTransition, while for clicking I used simply useSpring), but now I do not know how to combine these. I couldn't find any similar example. It is supposedly "easy" example, of a button with normal active and hover states which differ a bit. 

Anyone could help?

**Comment by joshuaellis** [comment]
Why do you feel the need to use `useTransition` and `useSpring` for one element? why not just use `useSpring` and the imperative API?

**Comment by robertlong** [comment]
I'm also having trouble with a similar situation. I'd like to use `useSprings` and `useTransition` together. I'm building an app where users can drag and rearrange tiles. They need to animate when they are dragged, dropped, reordered, and when items are added/removed. In order to accomplish this I'd like to use the imperative spring API for drag/drop in addition to the `useTransition` hook for add/remove/reordering. I'm not really sure how to accomplish this using the current API.

Right now I'm almost accomplishing it with `useSprings` and some custom deferred removal code, but it's not pretty and it's very brittle. I'd much rather use a well maintained and tested deferred removal method like `useTransition`.
